src/workerd/io/worker.c++

```diff
@@ -2414,66 +2484,52 @@ private:
   }
 };
 
-kj::Promise<void> Worker::Isolate::attachInspector(
-    kj::Timer& timer,
-    kj::Duration timerOffset,
-    kj::HttpService::Response& response,
-    const kj::HttpHeaderTable& headerTable,
-    kj::HttpHeaderId controlHeaderId) const {
-  KJ_REQUIRE(impl->inspector != nullptr);
-
-  kj::HttpHeaders headers(headerTable);
-  headers.set(controlHeaderId, "{\"ewLog\":{\"status\":\"ok\"}}");
-  auto webSocket = response.acceptWebSocket(headers);
-
-  return attachInspector(timer, timerOffset, *webSocket)
-      .attach(kj::mv(webSocket));
-}
-
 kj::Promise<void> Worker::Isolate::attachInspector(
     kj::Timer& timer,
     kj::Duration timerOffset,
     kj::WebSocket& webSocket) const {
   KJ_REQUIRE(impl->inspector != nullptr);
-
-  jsg::V8StackScope stackScope;
-  Isolate::Impl::Lock recordedLock(*this, InspectorChannelImpl::InspectorLock(nullptr), stackScope);
-  auto& lock = *recordedLock.lock;
-  auto& lockedSelf = const_cast<Worker::Isolate&>(*this);
-
   // If another inspector was already connected, boot it, on the assumption that that connection
   // is dead and this is why the user reconnected. While we could actually allow both inspector
   // sessions to stay open (V8 supports this!), we'd then need to store a set of all connected
   // inspectors in order to be able to disconnect all of them in case of an isolate purge... let's
   // just not.
-  lockedSelf.disconnectInspector();
+  //
+  // NB This method is only called from the main thread. If an earlier session was in the midst
```

Comment:
Isn't this called from the inspector thread, not the main thread?

Also note that in the internal codebase, there is no "main thread" nor is there an exclusive "inspector thread". This will be called from whatever thread received the inspector connection. Is it still safe to skip locks here?